cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -462,7 +468,29 @@ func (s *Server) updateRepo(ctx context.Context, repoRef *corev1.PackageReposito
 		}
 	}
 
-	repo.Spec.PassCredentials = auth != nil && auth.PassCredentials
+	repo.Spec.PassCredentials = request.Auth != nil && request.Auth.PassCredentials
+
+	// Get Flux-specific values
+	if request.CustomDetail != nil {
+		customDetail := &v1alpha1.FluxPackageRepositoryCustomDetail{}
+		if err := request.CustomDetail.UnmarshalTo(customDetail); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "customDetail could not be parsed due to: %v", err)
+		}
+		provider := customDetail.Provider
+
+		// following fixes for issue5746, the provider is allowed to be configured on update if not previously configured
+		if provider != "" && provider != "generic" {
+			if request.Auth != nil && request.Auth.Type != corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_UNSPECIFIED {
+				return nil, status.Errorf(codes.InvalidArgument, "Auth provider cannot be configured in combination with another auth method")
+			}
+			if repo.Spec.Provider != "" && repo.Spec.Provider != "generic" && repo.Spec.Provider != provider {
+				return nil, status.Errorf(codes.InvalidArgument, "Auth provider cannot be changed.")
+			}
+			repo.Spec.Provider = provider
+		} else {
+			repo.Spec.Provider = ""
+		}
```

Comment:
I'm just trying to understand some of the logic here: on lines 487 we explicitly disallow the provider from being changed if it has been set, but on line 491 we allow it to be set to empty. So, unless I'm missing something, users can change the auth provider, they just need to first set it to empty then set it?

Reply:
The idea was that changing the provider did not really make sense (the original code was completely ignoring the new value) so we are explicitly rejecting such change.
The main reason to allow to unset the provider is that the user may have tried with the provider and it did not work and as such is switching back to using basic auth for example.
But yes, a side effect would be that you could unset the provider and switch to another one in the next update.